src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -449,21 +448,28 @@ CodeWriter emitAndIndent(String s) throws IOException {
       if (trailingNewline) {
         emitIndentation();
         if (javadoc) {
-          out.append(" * ");
+          append(" * ");
         } else if (comment) {
-          out.append("// ");
+          append("// ");
         }
       }
-
-      out.append(line);
+      append(line);
       trailingNewline = false;
     }
     return this;
   }
 
-  private void emitIndentation() throws IOException {
+  private void emitIndentation() {
     for (int j = 0; j < indentLevel; j++) {
-      out.append(indent);
+      append(indent);
+    }
+  }
+
+  private void append(CharSequence csq) {
+    try {
+      out.append(csq);
+    } catch (IOException e) {
```

Comment:
Does that mean all `JavaFile.writeTo` methods would now throw `AssertionError`s instead of `IOException`s? Sounds bad.